src/SonarScanner.MSBuild.TFS/CoverageReportProcessorBase.cs

```diff
@@ -95,16 +109,20 @@ private bool TryConvertCoverageReports(IEnumerable<string> vscoverageFilePaths,
         {
             var xmlFileNames = vscoverageFilePaths.Select(x => Path.ChangeExtension(x, XmlReportFileExtension));
 
-            Debug.Assert(!xmlFileNames.Any(x => File.Exists(x)),
-                "Not expecting a file with the name of the binary-to-XML conversion output to already exist.");
-
-            var anyFailedConversion = vscoverageFilePaths.Zip(xmlFileNames, (x, y) => new { coverage = x, xml = y })
-                .Any(x => !this.converter.ConvertToXml(x.coverage, x.xml));
-
-            if (anyFailedConversion)
+            if (xmlFileNames.Any(x => File.Exists(x)))
+            {
+                Logger.LogWarning("Not expecting a file with the name of the binary-to-XML conversion output to already exist.");
+            }
```

Comment:
This logic doesn't seem quite right; if we hit this branch then we'll end up returning the whole list of expected _.coveragexml_ file names, even if only one of the expected files exist. Also, the function will return _true_.

If we reach this point, it means that the user has not supplied the `SonarProperties.VsCoverageXmlReportsPaths` property, but they have actually already done the conversion of some or all of the files from binary to xml.

We have a couple of options:
(a) we could treat this situation as if it were a failed conversion (i.e. return an empty list and `false`), or
(b) if there is an existing file, assume it has been converted correctly and just use it instead of trying to convert it.
What do you think?

Reply:
I had the b) in mind whilst changing the behavior, that's the reason why i did it like that, but yes you're right since we may consider that the folder might not be cleaned after each build.

Reply:
I hadn't thought about the folder not being cleaned between builds, but that's certainly another valid scenario.

I was thinking more that the code is making the potentially invalid assumption that if one file has already been converted by the user then all of the files have been converted.

Reply:
As discussed offline, let's fix this as a separate issue. I'll create a new ticket for it.

Reply:
FYI https://github.com/SonarSource/sonar-scanner-msbuild/issues/735 created.

Reply:
Thanks, closing this one.